src/component/Chat/ChatArea/ChatArea.js

```diff
@@ -3,44 +3,112 @@ import React, { useEffect, useState } from "react";
 import "./ChatArea.css"; // Import the CSS file
 import axios from "axios";
 
-const API = axios.create({ baseURL: `http://localhost:4500/` });
+const ENDPOINT = process.env.REACT_APP_ENDPOINT;
+const API = axios.create({ baseURL: ENDPOINT });
 
-const ChatArea = () => {
+const ChatArea = ({ receiverId, senderId, socket }) => {
   const [messageToSend, setMessageToSend] = useState("");
   const [messages, setMessages] = useState([]);
+  const [isSend, setIsSend] = useState();
+  const [conversationId, setConversationId] = useState();
+  const [reload, setReload] = useState(0);
+
   useEffect(() => {
     async function fetchMessage() {
       try {
-        const response = await API.get(
-          `/getmessage/${"64ed9d3cc28c30dc4ca83f0a"}`
+        const conversation = await API.get(
+          `/getallconversationofuser/${senderId}/${receiverId}`
         );
+        const fetchedConversation = conversation.data;
+        const conversationId = fetchedConversation._id;
+        setConversationId(conversationId);
+
+        const response = await API.get(`/getmessage/${conversationId}`);
         const fetchedMessage = response.data;
         setMessages(fetchedMessage);
       } catch (error) {
         console.error("Error fetching message:", error);
+        setMessages([]);
       }
     }
+
     fetchMessage();
-  }, []);
+  }, [senderId, receiverId, reload]);
+
+  useEffect(() => {
+    setIsSend(messageToSend.trim() !== "");
+  }, [messageToSend]);
+
+  const sendMessage = () => {
+    if (messageToSend.trim() === "") {
+      return;
+    }
+
+    const type = "text";
+    socket.emit("sendMessage", {
+      messageToSend,
+      type,
+      senderId,
+      conversationId,
+    });
+    setReload(reload + 1);
+    setMessageToSend("");
+    setIsSend(false);
+  };
 
   return (
     <div className="chat-area">
       <div className="messages">
-        {messages.map((msg, index) => (
-          <div key={index}>
-            <p>
-              <strong>{msg.message_data}: </strong>
-            </p>
-          </div>
-        ))}
+        {messages.length === 0 ? (
+          <p className="empty-messages">Let's chat with scroll</p>
+        ) : (
+          messages.map((msg, _id) => {
+            const messageDate = new Date(msg.createdAt);
+            const previousMessage = messages[_id - 1];
+            const previousMessageDate = previousMessage
+              ? new Date(previousMessage.createdAt)
+              : null;
+            const showDate =
+              !previousMessageDate ||
+              messageDate.toDateString() !== previousMessageDate.toDateString();
+
+            return (
+              <div key={_id}>
```

Comment:
@athar-shayan we discussed that we will not use indexed based Ids in office group

Reply:
this _id is use to get the pervious messages.